src/arch/cuda/hybridAPI/buddy_allocator.cpp

```diff
@@ -0,0 +1,195 @@
+#include "buddy_allocator.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
+#include <algorithm>
+#include <cuda_runtime.h>
+
+namespace buddy {
+  void allocator::print_status() {
+    printf("(buckets)\n");
+    size_t free = 0;
+    for (int i = 0; i < bucket_count; i++) {
+      printf("bucket[%d]: ", i);
+      for (const auto& block : buckets[i]) {
+        free += block.size;
+        printf("{%p, %lu} ", block.ptr, block.size);
+      }
+      printf("\n");
+    }
+
+    printf("(alloc_map)\n");
+    size_t allocated = 0;
+    size_t used = 0;
+    for (const auto& elem : alloc_map) {
+      const auto& block = elem.second;
+      allocated += block.size;
+      used += block.requested;
+      printf("ptr: %p, size: %lu, req: %lu\n", elem.first, block.size, block.requested);
+    }
+
+    printf("(fragmentation) free: %lu, allocated: %lu, used: %lu\n", free, allocated, used);
```

Comment:
Here and elsewhere, the format specifier for `size_t` is `%zu` rather than `%lu`.

Reply:
Good point, will fix them